src/core/RandomVariables.jl

```diff
@@ -38,20 +38,74 @@ export  VarName,
 ###########
 # VarName #
 ###########
+"""
+```
+struct VarName{sym}
+    csym      ::    Symbol
+    indexing  ::    String
+    counter   ::    Int
+end
+```
+
+A variable identifier. Every variable has a symbol `sym`, indices `indexing`, and 
+internal fields: `csym` and `counter`. The Julia variable in the model corresponding to 
+`sym` can refer to a single value or to a hierarchical array structure of univariate, 
+multivariate or matrix variables. `indexing` stores the indices that can access the 
+random variable from the Julia variable. 
+
+Examples:
+
+- `x[1] ~ Normal()` will generate a `VarName` with `sym == :x` and `indexing == "[1]"`.
+- `x[:,1] ~ MvNormal(zeros(2))` will generate a `VarName` with `sym == :x` and 
+ `indexing == "[Colon(), 1]"`.
+- `x[:,1][2] ~ Normal()` will generate a `VarName` with `sym == :x` and 
+ `indexing == "[Colon(), 1][2]"`.
+"""
 struct VarName{sym}
     csym      ::    Symbol        # symbol generated in compilation time
     indexing  ::    String        # indexing
     counter   ::    Int           # counter of same {csym, uid}
 end
+
+"""
+`VarName(csym, sym, indexing, counter)`
+`VarName{sym}(csym::Symbol, indexing::String)`
+
+Constructs a new instance of `VarName{sym}``
+"""
 VarName(csym, sym, indexing, counter) = VarName{sym}(csym, indexing, counter)
+function VarName(csym::Symbol, sym::Symbol, indexing::String)
+    # TODO: update this method when implementing the sanity check
+    VarName{sym}(csym, indexing, 1)
```

Comment:
Is it correct that the counter is always `1` in the constructor?

Reply:
I haven't changed this behavior. Maybe @xukai92 knows better.